app/src/main/java/org/piwigo/helper/URLHelper.java

```diff
@@ -0,0 +1,94 @@
+package org.piwigo.helper;
+
+import android.app.Activity;
+import android.app.ProgressDialog;
+import android.os.AsyncTask;
+import android.util.Log;
+
+import org.piwigo.ui.login.LoginActivity;
+import org.piwigo.ui.main.MainActivity;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.UnknownHostException;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLHandshakeException;
+import javax.net.ssl.SSLPeerUnverifiedException;
+
+public class URLHelper extends AsyncTask<String, Void, String> {
+
+    private AsyncUrlResponse delegate;
+
+    public URLHelper(AsyncUrlResponse delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    protected String doInBackground(String... url) {
+        String newUrl = url[0].replaceAll("https://", "").replaceAll("http://", "");
+        try {
+            if (isHttpsWebsite(newUrl))
+                return ("https://" + newUrl);
+            else if (isHttpWebsite(newUrl))
+                return ("http://" + newUrl);
+            else
+                return ("https://" + newUrl);
+        } catch (IOException e) {
+            Log.e("URLHelper", "IOException", e.getCause());
+            return ("https://" + newUrl);
+        }
+    }
+
+    @Override
+    protected void onPostExecute(String result) {
+        delegate.processFinish(result);
+    }
+
+    private boolean isHttpsWebsite(String siteUrl) throws IOException {
+        URL url = new URL("https://" + siteUrl);
```

Comment:
Any suggestions here ? I don't get your point.

Reply:
From an abstract point of view you "ask the URLHelper" by calling "isHttpWebsite" whether the given URL is serving HTTP requests with return code 200. So in case an IOException fires you could say: "no, that is not a valid http website" and return false. Somehow it is a bit asymmetric to catch SSLException and UnknownHostException but not the IOExcpetion. If this is intended, I propose to document that decision in a comment.

BTW: checking for response 200 might be already problematic, because is an URL not a valid httpWebsite if it returned 203 because there is a modifying proxy in between (I have no idea whether and how often that could happen) or some of the 3xx could be interesting to handle or 401 and 511... But I agree, that this is quite low prio and maybe we should just put this as a new ticket for the future...

Also in case of an UnknownHostException it might be good to not create the account for that site address, because we don't know whether it will work in future, even though it could be just because of network issues, but the given URL could be a perfectly valid piwigo address. So these subtle aspects are not clear (for me) if I read the method name "isHttpsWebsite". Maybe it would be better to call it "canConnectToHttpsWebsite"?

Another question that comes to my mind is: How can we guide the user in case just mistyped the path of the address, but the server perfectly responds to a HTTPS request... Is that a "HttpsWebsite" - in a sense yes, just the path was wrong. Maybe the HTTP/HTTPS check should be done only based on the host name, completely ignoring the path? - But then what happens if the server serves a piwigo instance at the given path for HTTPS and on HTTP it serves a completely different static web site?

Interesting is also what happens when I enter "piwigo.org" right now...

Long story, I know, so don't get me wrong we don't need to answer all the questions I raised to the end now, I mainly want to recommend is to try to be precise, consistent and use intuitive names and document a little bit more about the intended functionality. Maybe in a few English words it gets clear what the caller of this function should expect and whether it is the responsibility of the caller or this function to check for those aspects - in case we go for that level of detail in a few releases.